### DISCUSSION TITLE
[Feature] The ability to batch change a recognised face to another

### DISCUSSION THREAD

**Discussion by edoardo-c** [immich-app/immich#12460] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This is a feature request to be able to do batch changes to face selections.

Say I have two children, 
- `John (2000)`
- `Paul (2003)` 

For whatever reason ML or I got the faces mixed up. I'd like to be able to search for Paul's faces between 2000 and 2002 and from the result, be able to replace the face common to all images and replace it with John.

This is very handy for large libraries where a mistake was made which impacts hundreds of images.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by xgoisern** [comment]
+1, I am in a similar situation, and batch changes to face assignements would be very usefull!

**Reply by pyccl** [reply]
+1

**Comment by martabal** [comment]
You can already do that with the « fix incorrect match » button

**Reply by edoardo-c** [reply]
Could you tell me where the button is so that I can try it out? I set the face and dates.  then select images but I see no button. What is your process to get there?

**Reply by martabal** [reply]
Go to `Explore` &rarr; `People` &rarr; Go to the person page with the incorrect faces &rarr; Select the assets with the incorrect faces &rarr; Click on the three dots button &rarr; Click on "fix incorrect match"

**Reply by edoardo-c** [reply]
@martabal , thanks for trying to help but this does not solve my problem. The problem I have is with PersonA being identified as PersonB at a time where PersonA was not yet born. As of birth date, I think most face tags are correct but there is no way for me to do a filter that says: Look for PersonA between two dates and relabel as PersonB. In short, a mass relabel.

I expect the error comes when you have brothers/sisters with family traits, etc. In my case, I have over 250 images with the wrong face label and I am not ready to do the changes one by one ;)

**Reply by christerdk** [reply]
> @martabal , thanks for trying to help but this does not solve my problem. The problem I have is with PersonA being identified as PersonB at a time where PersonA was not yet born. As of birth date, I think most face tags are correct but there is no way for me to do a filter that says: Look for PersonA between two dates and relabel as PersonB. In short, a mass relabel.
> 
> I expect the error comes when you have brothers/sisters with family traits, etc. In my case, I have over 250 images with the wrong face label and I am not ready to do the changes one by one ;)

This is exactly what I'm experiencing now. Our most recent newborn was added as face to our firstborn (who is now 6YO). I guess this happened because when I started with Immich, we added the most recent photos first and then I worked backwards. So the second son was already in the system when the photos of our first were added. 

Batch change Face A to Face B in search result, much like batch change location etc, is definitely needed.

**Reply by bnavigator** [reply]
Had the same issue. You can select all 250 images at once and "re-assign".

**Reply by Ashtowake** [reply]
> Had the same issue. You can select all 250 images at once and "re-assign".

Would you mind elaborating how you re-assign? I have selected all my images (also around 200) through the search feature and want to change the recognised person to a different one, but there is no re-assign option.

**Reply by bnavigator** [reply]
You select the pictures, go to the three-dot-menu button on the right and select "Fix incorrect match". Then you reassign.

**Reply by Ashtowake** [reply]
> You select the pictures, go to the three-dot-menu button on the right and select "Fix incorrect match". Then you reassign.

I know about that feature but sadly it's only available in the People view. If I search for assets anywhere else and select incorrectly recognised faces, I am not given that option. I had hoped there was a way I simply missed because this requires me to manually select the images in the People view, where no further search refinement is possible, instead of simply being able to open the album they are already correctly sorted in, Ctrl+A and re-assign.
